### PR TITLE
create medication-refill.md

### PR DESCRIPTION
**Pull request by isaacvetter**

### CODE REVIEW

**Comment by mattvarghese** [comment]
Hey @kpshek, @jmandel, @brettmarquard and @brynrhodes:
Is there any concerns around publishing this hook for evaluating decision support upon a pharmacy/patient/user requesting a medication refill on the patient's behalf? From experience, I would think that this is a valuable decision support point.
Could you guys please review this pull request?

File: docs/hooks/medication-refill.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+## Context
+
+The set of medications proposed or in progress of being prescribed. All FHIR resources in this context MUST be based on the same FHIR version. All FHIR resources in the medications object MUST have a status of _draft_.
+
+Field | Optionality | Prefetch Token | Type | Description
+----- | -------- | ---- | ---- | ----
+`userId` | OPTIONAL | Yes | *string* | The id of the current user.<br />For this hook, the user is expected to be of type [Practitioner](https://www.hl7.org/fhir/practitioner.html).<br />For example, `Practitioner/123`
+`patientId` | REQUIRED | Yes | *string* |  The FHIR `Patient.id` of the current patient in context
+`encounterId` | OPTIONAL | Yes | *string* |  The FHIR `Encounter.id` of the current encounter in context
+`medications` | REQUIRED | No | *object* | DSTU2 - FHIR Bundle of _draft_ MedicationOrder resources <br/> STU3 - FHIR Bundle of _draft_ MedicationRequest resources
```
Comment (brynrhodes) [inline docs/hooks/medication-refill.md]: Add R4 specification

File: docs/hooks/medication-refill.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+### Example (STU3)
```
Comment (brynrhodes) [inline docs/hooks/medication-refill.md]: Add R4 example

Comment (dloewens) [reply docs/hooks/medication-refill.md]: @brynrhodes In the R4 example, the MedicationRequest has:
                  "status":"draft",
                  "intent":"order",

I was curious as to the status to expect in a MedicationRequest for an order generated by a Refill Authorization Request (RAR): 
http://community.fhir.org/t/medicationrequest-status-and-refill-authorization-requests/1628 

Lloyd responded to the post indicating he thought that a new RAR would either be a Task resource or if it were a MedicationRequest resource then it wold be a MedicationRequest resource with: 
   "status":"active" 
   "intent":"proposal". 
Is Lloyd's response compelling enough to update the example? Alternatively, is this something that ought to be taken to the pharmacy work group for discussion?

**Comment by isaacvetter** [comment]
Approved by CDS wg during today's call for merge.

**Pull request merged**